[IMP] sales: update orders and variants for 16 (#5456)
odoo/documentation: content/applications/sales/sales/send_quotations/orders_and_variants.rst

.. image:: orders_and_variants/order-grid-entry-setting.png
   :align: center
   :alt: The variant grid entry setting in the Odoo Sales application

.. note::
   Of course, the :guilabel:`Variants` feature **must** also be activated, as well, in order to use
   product variants on quotations and sales orders.

Product configuration
=====================

Once the :guilabel:`Order Grid Entry` setting is enabled, both options (:guilabel:`Product
Configurator` and :guilabel:`Order Grid Entry`) become available on every product form.

To configure a product form to use either a product configurator or variant grid entry, start by
navigating to :menuselection:`Sales app --> Products --> Products` to view all the products in the
database.

Then, select the desired product to configure, or click :guilabel:`New` to create a new product from
scratch. Once on the product form, click into the :guilabel:`Attributes & Variants` tab, where
product variants can be viewed, modified, and added.

At the bottom of the :guilabel:`Attributes & Variants` tab, there is a :guilabel:`Sales Variant
Selection` section with two options: :guilabel:`Product Configurator` and :guilabel:`Order Grid

> Entry`.

Reviewer: It probably seems self-explanatory, but could be worth a call out that these options will only appear if attributes have been added to the record.

===============================================
Product variants on quotations and sales orders
===============================================

Before getting into detail about how to use product variants on quotations and sales orders, it's
recommended to learn about :doc:`/applications/sales/sales/products_prices/products/variants` in
Odoo.

Once familiarized with the basics surrounding product variants, the following covers how product
variants can be added to quotations and sales orders using the *Product Configurator* or *Order Grid
Entry*.

.. note::
   It should be noted that the setting is titled, `Variant Grid Entry` on the *Sales* app settings

>    page, and titled, `Order Grid Entry` on product forms. So, be sure to keep that in mind.

Reviewer 2: Just wondering why the "Order Grid Entry" and "Variant Grid Entry" features are surrounded with the back-tick (`) here? Is it to make these feature names stand out in the note admonition block?

Was just wondering about your rationale for this formatting choice, since we use guilabels when referring to UI items (not what you're doing here), [italics for concepts](https://docs.google.com/document/d/1laiWC9ow0IjXPuQp5h8L_Tj98HbE8ymGn9v-FFOwlYM/edit#heading=h.90nibixh8llr), [back-ticks for user entries](https://docs.google.com/document/d/1laiWC9ow0IjXPuQp5h8L_Tj98HbE8ymGn9v-FFOwlYM/edit#heading=h.ipar7km38ci3)

Just wondering how consistent should we be in our formatting choices? Also when you get to this comment, can you let me know your thoughts in a Discord DM? I want to know what you think, and I miss these github conversations a lot

Author: Great question. So, my initial intent was to use ticks because these elements aren't directly on the screen that's being explained. however, I do believe you're correct -- these should be GUI, so I'll make that change right away. Thanks for pointing that out.

Settings
========

When working with product variants, Odoo uses the :guilabel:`Product Configurator`, by default. To
add the :guilabel:`Variant Grid Entry` option, that feature **must** be enabled in the Odoo *Sales*

> application.

Reviewer 2: A brief explanation about why we'd want to add the Variant Grid Entry option here would help readers decide whether they want to add this feature :open_mouth:

To enable that setting, go to :menuselection:`Sales app --> Configuration --> Settings`, and scroll
to the :guilabel:`Product Catalog` section. Then, check the box next to the :guilabel:`Variant Grid
Entry` option, and click :guilabel:`Save`.

.. image:: orders_and_variants/order-grid-entry-setting.png
   :align: center

>    :alt: The variant grid entry setting in the Odoo Sales application

Reviewer 2: ```suggestion
   :alt: The variant grid entry setting in the Odoo Sales application.
```

.. note::

>    Of course, the :guilabel:`Variants` feature **must** also be activated, as well, in order to use
>    product variants on quotations and sales orders.

Reviewer 2: ```suggestion
   Of course, the :guilabel:`Variants` feature **must** also be activated, in order to use product
   variants on quotations and sales orders.
```

Product configuration
=====================

Once the :guilabel:`Variant Grid Entry` setting is enabled, both options (:guilabel:`Product
Configurator` and :guilabel:`Order Grid Entry`) become available on every product form.

To configure a product form to use either a product configurator or variant grid entry, start by
navigating to :menuselection:`Sales app --> Products --> Products` to view all the products in the
database.

> Then, select the desired product to configure, or click :guilabel:`New` to create a new product from
> scratch. Once on the product form, click into the :guilabel:`Attributes & Variants` tab, where

Reviewer 2: ```suggestion
Then, select the desired product to configure, or click :guilabel:`New`, to create a new product
from scratch. Once on the product form, click into the :guilabel:`Attributes & Variants` tab, where
```

product variants can be viewed, modified, and added.

At the bottom of the :guilabel:`Attributes & Variants` tab, there is a :guilabel:`Sales Variant
Selection` section with two options: :guilabel:`Product Configurator` and :guilabel:`Order Grid
Entry`.

> .. note::
>    It should be noted that these options **only** appear if attributes have been added to the
>    record.

Reviewer 2: To add to Jess' earlier comment, I tested adding 2 attributes with 1 value each, and the two Sales  Variant Selection options were still not available
![image](https://github.com/odoo/documentation/assets/23611711/444c1c5a-432a-4b3f-a14d-dee422e05261)

```suggestion
.. note::
   It should be noted that these options **only** appear if at least two values of an attribute have
   been added to the record.
```

   been added to the record.

.. image:: orders_and_variants/attributes-variants-tab-selection-options.png
   :align: center
   :alt: Sales variant selection options on the attributes and variants tab on product form.

These options determine which method is used when adding product variants to quotations or sales
orders.

The :guilabel:`Product Configurator` provides a pop-up window that neatly displays all the available
product variants for that particular product when it's added to a quotation. However, only one
variant can be selected/added at a time.

The :guilabel:`Order Grid Entry` provides the same information as the :guilabel:`Product
Configurator` in a table layout, allowing the user to select larger numbers of unique product
variants, and add them to a quotation/sales order, in a single view.

.. note::
   The product must also have at least two variants (one attribute with two values) created in order
   for the :guilabel:`Sales Variant Selection` option to appear. Click :guilabel:`Add a line` under
   :guilabel:`Attribute` to add attributes and values to this product and create product variants.

Product configurator
====================

> The :guilabel:`Product Configurator` feature appears as a :guilabel:`Configure` pop-up window, as
> soon as a product with (at least two) variants is added to a quotation or sales order, but **only**
> if the :guilabel:`Product Configurator` option is selected on its product form.

Reviewer 3: When referring to the feature no guilabel is needed, but when referring to the option field on the UI guilabel is needed
```suggestion
The product configurator feature appears as a :guilabel:`Configure` pop-up window, as soon as a
product with (at least two) variants is added to a quotation or sales order, but **only** if the
:guilabel:`Product Configurator` option is selected on its product form.
```
